### PR TITLE
test(rest): added more event tests

### PR DESCRIPTION
**Pull request by rapaktech**
Signed-off-by: Jim Ezesinachi <jim@animo.id>

### CODE REVIEW

**Comment by rapaktech** [comment]
Don't merge yet. I'm wrapping up the update to the documentation and CLI

**Comment by rapaktech** [comment]
Looks like Prettier is failing on the README

**Comment by TimoGlastra** [comment]
You can fix it using yarn format

File: packages/rest/README.md [hunk 1]
```diff
@@ -140,26 +140,31 @@ const run = async () => {
...
 
 - `Basic messages`
 - `Connections`
 - `Credentials`
 - `Proofs`
 
-When using the CLI a webhook url can be specified using the `--webhook-url` config option. When using the rest server as an library the webhook url can be configured in the `startServer` method.
+When using the CLI, a webhook url can be specified using the `--webhook-url` config option.
+
+When using the REST server as an library, the WebSocket server and webhook url can be configured in the `startServer` method.
```
Comment (TimoGlastra) [inline packages/rest/README.md:158]: Maybe we can add a bit more context to what the different methods do:
- setupServer will not automatically create the socket server, even if no default is provided
- startServer will create and start a web socket server on the http port if no server is provided, and will use the provided server if available

File: packages/rest/tests/proof.test.ts [hunk 2]
```diff
@@ -238,10 +240,47 @@ describe('ProofController', () => {
...
+        type: ProofEventTypes.ProofStateChanged,
+        payload: {
+          previousState: null,
+          proofRecord,
+        },
+      })
+
+      // Wait for event on WebSocket
+      const event = await waitForEvent
+      expect(event).toHaveProperty('type', 'ProofStateChanged')
```
Comment (TimoGlastra) [inline packages/rest/tests/proof.test.ts:275]: Maybe we can add a test for the structure so we know the json was serialized correctly and all fields are present?

**Review by TimoGlastra — APPROVED** [review]
Some bits, but looks good

File: packages/rest/README.md (hunk 1 above)
Comment (rapaktech) [reply packages/rest/README.md:158]: On it

File: packages/rest/tests/proof.test.ts (hunk 2 above)
Comment (rapaktech) [reply packages/rest/tests/proof.test.ts:275]: On it

**Pull request merged**